apisix/init.lua

```diff
@@ -325,8 +325,29 @@ function _M.http_access_phase()
 
     core.ctx.set_vars_meta(api_ctx)
 
+    local uri = api_ctx.var.uri
+    if local_conf.apisix and local_conf.apisix.delete_uri_tail_slash then
+        if str_byte(uri, #uri) == str_byte("/") then
+            api_ctx.var.uri = str_sub(api_ctx.var.uri, 1, #uri - 1)
+            core.log.info("remove the end of uri '/', current uri: ",
+                          api_ctx.var.uri)
+        end
+    end
+
+    local matched_internal_api = false
+    if router.api.has_route_not_under_apisix() or
+        core.string.has_prefix(uri, "/apisix/")
+    then
+        matched_internal_api = router.api.match(api_ctx)
```

Comment:
The internal API will call `exit` inside so there is not change to execute the global rule.
Maybe we can refactor and exact the global rule code, then run it before https://github.com/apache/apisix/blob/c895cdf4ff008c13dec8197e9f0a0407be144517/apisix/api_router.lua#L124

Reply:
got it, thanks!